texastribune/donations
# Dynamic donation suggestion

#### What's this PR do?
Adds functionality to automatically divide the yearly suggested donation amount by 10 when switching to the monthly option and to multiply the monthly by 10 when switching to the yearly option.

#### Why are we doing this? How does it help us?
We're hoping to simplify the donor process and thus bring in more (mostly monthly) donors.

#### How should this be manually tested?
* Spin up locally
* Visit /donate and make sure the amount doesn't change when switching between monthly and yearly
* Visit "/donate?installmentPeriod=yearly&amount=200&abTesting=True"
* Ensure the monthly amount is 20 and the yearly and one-time amounts are 200
* Visit "/donate?installmentPeriod=monthly&amount=15&abTesting=True"
* Ensure the monthly amount is 15 and the yearly and one-time amounts are 150
* Ensure that changing the amount breaks this automation
    * i.e. switching between monthly/yearly doesn't effect the amount

#### How should this change be communicated to end users?
We'll let membership know.

#### Are there any smells or added technical debt to note?
@ashley-hebler, you had suggested providing an amount for the alternative (vs. automatically dividing/multiplying by 10) but I went with this more automatic route for now since that was the initial intention. I'll talk to membership about the other option (and we could even do both where we default to this automatic way UNLESS there is an alternative value given).

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recAbvCibzXZjDT6G?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
